### ISSUE TITLE
Text test reads from invalid memory and has invalid free/delete bug

### ISSUE THREAD

**Issue by sebastianbaginski** [cogneco/ooc-kean#785]
Something is wrong with the `operator == (Text, Text)` method in Text class, it causes invalid memory reads.
To reproduce, run
```
./test.sh base/Text
```
and run valgrid on `Tests` executable.
Invalid free/delete is in the same method.

**Comment by sebastianbaginski** [comment]
```
// class Text
	operator == (other: This) -> Bool {
		result := this _buffer == other _buffer
		if (this _buffer raw != other _buffer raw)
			other free(Owner Receiver)
		this free(Owner Receiver)
		result
	}

// class TextBuffer:
	raw: Char* { get {
		result := this _backend pointer as Char*
		this free(Owner Receiver)
		result
	}}
```
Looks like `if (this _buffer raw != other _buffer raw)` already calls ` free(Owner Receiver)` on both objects. @simonmika

**Comment by tomasmahlberg** [comment]
@simonmika Are you looking at this or do you want me to take a look? It should be an easy fix. I think @davidhesselbom wants this into upstream ASAP as it is currently a bit broken.

**Comment by tomasmahlberg** [comment]
So, I took the liberty to fix this: #803.
